web/src/user_topics.ts

```diff
@@ -15,7 +16,32 @@ import * as timerender from "./timerender";
 import * as ui_report from "./ui_report";
 import {get_time_from_date_muted} from "./util";
 
-const all_user_topics = new Map();
+type ServerUserTopic = z.infer<typeof user_topic_schema>;
```

Comment:
Does it make sense to have this lione be just after the `const user_topic_schema` line? That approach seems easier to read to me at least.

Reply:
I'm going not block on this one for merging because I think we have this pattern in like 5 other files using `zod`, so we should just change them in a sweep if we decide to do so. Might be better to discuss in a #frontend thread anyway; perhaps @Lalit3716 you can start one.